packages/react-reconciler/src/ReactFiberLane.js

```diff
@@ -89,11 +86,8 @@ const InputContinuousLanes: Lanes = /*                  */ 0b0000000000000000000
 export const DefaultHydrationLane: Lane = /*            */ 0b0000000000000000000000100000000;
 export const DefaultLanes: Lanes = /*                   */ 0b0000000000000000000111000000000;
 
-const TransitionShortHydrationLane: Lane = /*           */ 0b0000000000000000001000000000000;
-const TransitionShortLanes: Lanes = /*                  */ 0b0000000000000011110000000000000;
-
-const TransitionLongHydrationLane: Lane = /*            */ 0b0000000000000100000000000000000;
-const TransitionLongLanes: Lanes = /*                   */ 0b0000000001111000000000000000000;
+const TransitionHydrationLane: Lane = /*                */ 0b0000000000000000001000000000000;
```

Comment:
Is the hydration lane used for loads or is it used for something else?

Reply:
Hydration lanes are a used for this scenario:

- During progressive hydration, only part of the initial HTML has been hydrated. Let's say a parent is hydrated, but its child has not.
- Then, the parent receives an update (`setState`) and attempts to re-render the child. But the child hasn't even hydrated yet; it's still just showing the initial HTML.
- We could render a brand new child using the newest props, but that would require  discarding the initial HTML and losing its state, like text input.
- So instead we do a clever thing: pause the parent update, then bump the priority of hydrating the child to a slightly higher priority.
- Finish hydrating the child. It has now fully mounted.
- Now try the parent update again and proceed like normal.

In order for the "bump the priority" trick to work, we reserve special lanes that sit in between the ranges used for regular updates. The hydration lanes only ever include this type of work, so that they don't get accidentally batched together with other updates.

Reply:
See: https://github.com/facebook/react/blob/60ba723bf78b9a28f60dce854e88e206fab52301/packages/react-dom/src/__tests__/ReactDOMServerPartialHydration-test.internal.js#L506-L575

Reply:
Awesome, that makes sense. Is it possible for hydration lanes to get entangled with non-hydration lanes?

Reply:
I think theoretically it could happen with `useMutableSource`, which uses entanglement to de-opt and prevent tearing in some cases when it detects a mutation.